core/src/main/java/tc/oc/pgm/spawner/SpawnerModule.java

```diff
@@ -97,13 +146,27 @@ public SpawnerModule parse(MapFactory factory, Logger logger, Document doc)
                 minDelay,
                 maxDelay,
                 maxEntities,
-                numericID);
+                numericId);
```

Comment:
id should be the first param for spawner definition, and should come from a player-defined attribute if one is available, otherwise, use an autogenerated one
spawner definition, if it doesn't already, should extend SelfIdentifyingFeatureDefinition
You can take inspiration for the autogenerated id stuff from "PostIdSerial" and `PostDefinition.makeDefaultId` https://github.com/PGMDev/PGM/pull/984/files#diff-7aa0581d0bce176dc53a9eadaca6d5a541de970854921dbe6b302cd51f542807R81 (keep in mind that this example creates an ID from a NAME, which, neither composite posts nor spawners have names, which is why first param is always null)